### PR TITLE
Support ENUM type in parquet reader

### PR DESCRIPTION
**Pull request by qqibrow**
This PR adds ENUM type support in parquet and fixes logging

### CODE REVIEW

**Comment by qqibrow** [comment]
will wait for #8000 to merge and then rebase

File: velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp [hunk 1]
```diff
@@ -892,3 +892,30 @@ TEST_F(ParquetReaderTest, prefetchRowGroups) {
...
+
+  facebook::velox::dwio::common::ReaderOptions readerOptions{leafPool_.get()};
+  auto reader = createReader(sample, readerOptions);
+  EXPECT_EQ(reader->numberOfRows(), 3ULL);
+
+  auto rowType = reader->typeWithId();
+  EXPECT_EQ(rowType->type()->kind(), TypeKind::ROW);
+  EXPECT_EQ(rowType->size(), 1ULL);
+
+  auto varcharType = rowType->childAt(0);
```
Comment (yingsu00) [inline velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp:908]: Just inline this varcharType in the following line

File: velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp [hunk 2]
```diff
@@ -892,3 +892,30 @@ TEST_F(ParquetReaderTest, prefetchRowGroups) {
...
+  EXPECT_EQ(varcharType->type()->kind(), TypeKind::VARCHAR);
+
+  auto fileSchema = ROW({"test"}, {VARCHAR()});
+  auto rowReaderOpts = getReaderOpts(fileSchema);
+  auto scanSpec = makeScanSpec(fileSchema);
```
Comment (yingsu00) [inline velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp:913]: inline scanSpec

**Review by mbasmanova — COMMENTED** [review]
@qqibrow CI is red. Please take a look and rebase.

**Comment by qqibrow** [comment]
@yingsu00 @mbasmanova all tests pass. please take a look. thanks!

**Review by mbasmanova — APPROVED** [review]
@qqibrow Thanks. Would you confirm that the new test fails without the code changes?

**Review by xiaoxmeng — APPROVED** [review]

**Comment by qqibrow** [comment]
> @qqibrow Thanks. Would you confirm that the new test fails without the code changes?

yes.

**Comment by mbasmanova** [comment]
@qqibrow Thank you for the confirmation.

**Pull request closed without merging**